### PR TITLE
Project 1.4

### PR DESCRIPTION
**Pull request by JasonGibson274**
This contain a write up and simple exercises where the students will run a google Colab notebook in order to train a simple NN on the MNIST dataset. They will then play around with some parameters (learning rate and epochs). Finally they apply it to KMNIST.

I think overall this is a little short.

### CODE REVIEW

File: assignments/week_1/project1.4/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Project 1.4
+Welcome to this first robotics themed projects of the RoboJackets software training
+curriculum. The point of these projects is to give you hands on experience playing
+around with code related to a robotics topic that we have covered in lecture.
+I hope you enjoy this one
+
+# Project Objective
+During this project we will be training a Neural Network on the MNIST dataset.
+This is the canonical hello world of the machine learning world. The goal is to
+classify images of handwritten digits into what it is. We will be using a
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: `The goal is to classify images of handwritten digits into the digits they represent`?

File: assignments/week_1/project1.4/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+convolutional neural network in order to effectively process the image data.
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: Tie in to convolutions project!

File: assignments/week_1/project1.4/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+## Python Notebooks Basics
+
+Each block is of the same code, just press the play button to run that section of code.
+The results from each section are kept in memory, this allows you to run a further
+along section of the code multiple times without constantly rerunning the initial sections.
+If you change a section above another you will have to rerun it.
+
+## Part 1
+For this section you should run the entire notebook step by step. I would recommend
+using the button at the top pictured here in order to make it easier on yourself.
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: Pictured where?

File: assignments/week_1/project1.4/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+You should see output similar to this, the instructions inside the notebook should
+cover any questions you might have about what each section of code is doing.
+You should not read over each line of code unless instructed to do so, you won't
+need to know how any of the code works in order to complete the exercise (I wrote
+it and I barely know how it works).
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: This does not ~spark joy~ inspire confidence.

File: assignments/week_1/project1.4/ReadMe.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+You should see a graph similar to the one below at the final output.
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: ??

File: assignments/week_1/project1.4/ReadMe.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+## Part 2
+Now that we can train a simple model we want to try to increase out accuracy. Remember
+how we discussed underfitting the model previously. Now our model has enough room to
+accurate model the data (trust me on this one). So what else could we look at in order to
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: Describe this first dataset

Comment (kylestach) [reply assignments/week_1/project1.4/ReadMe.md]: And also maybe describe the model

File: assignments/week_1/project1.4/ReadMe.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+cover any questions you might have about what each section of code is doing.
+You should not read over each line of code unless instructed to do so, you won't
+need to know how any of the code works in order to complete the exercise (I wrote
+it and I barely know how it works).
+
+You should see a graph similar to the one below at the final output.
+
+## Part 2
+Now that we can train a simple model we want to try to increase out accuracy. Remember
+how we discussed underfitting the model previously. Now our model has enough room to
```
Comment (kylestach) [inline assignments/week_1/project1.4/ReadMe.md]: Uh, where did you discuss this?

Also, maybe it would be good to give people intuition for over/underfit by fitting a polynomial to some data. Something like this?

image

**Review by kylestach — CHANGES_REQUESTED** [review]

File: assignments/week_1/project1.4/ReadMe.md (hunk 7 above)
Comment (JasonGibson274) [reply assignments/week_1/project1.4/ReadMe.md]: You keep having better graphics than me. I'm impressed and happy.

File: assignments/week_1/project1.4/ReadMe.md (hunk 4 above)
Comment (JasonGibson274) [reply assignments/week_1/project1.4/ReadMe.md]: I think it gets the point across of you don't need to understand the entire script.

**Review by kylestach — APPROVED** [review]
